lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/REEFEventHandlers.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runime.azbatch.driver;
+
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.runtime.common.driver.api.RuntimeParameters;
+import org.apache.reef.runtime.common.driver.resourcemanager.NodeDescriptorEvent;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceAllocationEvent;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceStatusEvent;
+import org.apache.reef.runtime.common.driver.resourcemanager.RuntimeStatusEvent;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.wake.EventHandler;
+
+import javax.inject.Inject;
+
+/**
+ * Helper that represents the REEF layer to the Azure Batch runtime.
+ */
+// This is a great place to add a thread boundary, should that need arise.
+@Private
+public final class REEFEventHandlers implements AutoCloseable {
```

Comment:
Not sure about the name being plural. This is usually saved to denote a list or collection. Also, we should avoid using "REEF" in the class name. Although they've been inconsistent in the reef project, acronyms should be camel case. I suggest "ReefEventHandler".

Reply:
I am fine with either. I just prefer consistency. I saw more uses of REEF as opposed to Reef in the code. Also, there are "REEFEventHandlers" classes in the other runtimes. So it makes sense to keep it the same - just for consistency.